packages/gatsby-plugin-sitemap/src/internals.js

```diff
@@ -1,123 +1,119 @@
-import fs from "fs"
-import pify from "pify"
-import minimatch from "minimatch"
+import reporter from "gatsby-cli/lib/reporter"
 
+/**
+ *
+ * @param {string} path
+ * @returns {string}
+ */
 export const withoutTrailingSlash = path =>
   path === `/` ? path : path.replace(/\/$/, ``)
 
-export const writeFile = pify(fs.writeFile)
-export const renameFile = pify(fs.rename)
-
-export function filterQuery(
-  results,
-  excludes,
-  pathPrefix,
-  resolveSiteUrl = defaultOptions.resolveSiteUrl
-) {
-  const { errors, data } = results
-
-  if (errors) {
-    throw new Error(errors.join(`, `))
-  }
-
-  const { allSitePage, ...otherData } = data
-
-  let { allPages, originalType } = getNodes(allSitePage)
-
-  // Removing excluded paths
-  allPages = allPages.filter(
-    page =>
-      !excludes.some(excludedRoute =>
-        minimatch(
-          withoutTrailingSlash(page.path),
-          withoutTrailingSlash(excludedRoute)
-        )
-      )
-  )
-
-  // Add path prefix
-  allPages = allPages.map(page => {
-    page.path = (pathPrefix + page.path).replace(/^\/\//g, `/`)
-    return page
-  })
-
-  // siteUrl Validation
+/**
+ * @name prefixPath
+ *
+ * Properly handles prefixing relative path with site domain, Gatsby pathPrefix and AssetPrefix
+ *
+ * @param {string} url - string containing relative path
+ * @param {string} siteUrl - results of the resolveSiteUrl function
+ * @returns {string}
+ */
+// TODO: Update for v3
+export function prefixPath({ url, siteUrl, pathPrefix = `` }) {
+  return new URL(pathPrefix + url, siteUrl).toString()
+}
 
-  let siteUrl = resolveSiteUrl(data)
+/**
+ * @name resolveSiteUrl
+ *
+ * @param {object} data - results of the GraphQL query
+ * @returns {string} - site URL, this can come from thegraphql query or another scope
+ */
+export function resolveSiteUrl(data) {
+  return data.site.siteMetadata.siteUrl
+}
+/**
+ * @name resolvePagePath
+ *
+ * if you don't want to place the URI in "path" then resolvePagePath
+ * are needed.
+ *
+ * @param {object} page - Array Item returned from resolvePages
+ * @returns {string} - uri of the page without domain or protocol
+ */
+
+export function resolvePagePath(page) {
+  return page?.path
```

Comment:
What happens if page.path is null? should we throw / warn the user?
```suggestion
  return page?.path
```